BlueDBOnDisk/src/main/java/io/bluedb/disk/BlueDbOnDisk.java

```diff
@@ -81,6 +81,12 @@ public Path getPath() {
 
 	@SuppressWarnings({"rawtypes", "unchecked"})
 	protected BlueCollectionOnDisk getCollection(String folderName) throws BlueDbException {
-		return new BlueCollectionOnDisk(this, folderName, Serializable.class);
+		synchronized (collections) {
+			BlueCollectionOnDisk collection = collections.get(folderName);
+			if (collection == null) {
+				collection = new BlueCollectionOnDisk(this, folderName, Serializable.class);
+			}
+			return collection;
+		}
```

Comment:
If you look at the other getCollection method this isn't doing everything it needs to. For example, its not adding the collection to the map. What is this one used for? Is it only for tests?

Reply:
It's only for backup.  Although perhaps it should add to the map to remove the chances of a duplicate collection being created

Reply:
Actually, if we put it in the map, it will have the type Serializable which will throw exceptions if we try to get it with another type later.